docs/filters.rst

```diff
@@ -5,12 +5,33 @@ Filtering
 .. py:module:: web3.utils.filters
 
 
-The :meth:`web3.eth.Eth.filter` method can be used to setup filter for:
+The :meth:`web3.eth.Eth.filter` method can be used to setup filters for:
+
+* Pending Transactions: ``web3.eth.filter('pending')``
+
+* New Blocks ``web3.eth.filter('latest')``
 
-* Pending Transactions
-* New Blocks
 * Event Logs
 
+    Through the contract instance api:
+
+    .. code-block:: python
+
+        event_filter = myContract.eventFilter('eventName', {'filter': {'arg1':10}})
+
+    Or built manually by supplying `valid filter params <http://https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_newfilter/>`_ params:
+
+    .. code-block:: python
+
+        event_filter = web3.eth.filter({"address": contract_address})
+
+* Attaching to an existing filter
+
+    .. code-block:: python
+
+        from web3.auto import w3
+        existing_filter = web3.utils.filters.Filter(w3, "0x0")
```

Comment:
If we're going to document manual instantiation of `Filter` objects, then the class needs to be moved from within the `web3.utils` directory since `web3.utils` is explicitly non-public API.

Reply:
What about adding an option to select a filter by `filter_id` to `Eth.filter`?

Reply:
I like it.

Reply:
https://github.com/ethereum/web3.py/pull/641